RawVideo/RawVideo/ContentView.swift

```diff
@@ -0,0 +1,1032 @@
+        selectedVideoDeviceInfoIndex = videoDeviceInfoList.count > 0 ? 0 : -1;
+        
+        for i in 0 ..< videoDeviceInfoList.count
+        {
+            let videoDeviceInfoItem = VideoDeviceInfoItem(id: i, name: videoDeviceInfoList[i].name)
+            videoDeviceInfoItemList.append(videoDeviceInfoItem)
+        }
+        
+        cameraList = CameraCaptureService.GetCameraList()
+        cameraItemList = [CameraItem]();
+        selectedCameraIndex = cameraList.count > 0 ? 0 : -1;
+        
+        for i in 0 ..< cameraList.count
+        {
+            let cameraItem = CameraItem(id: i, name: cameraList[i].localizedName)
+            cameraItemList.append(cameraItem)
+        }
+        
+        showCallSettings = true
+        
+        let defaults = UserDefaults.standard
+        let savedMeetingLink = defaults.value(forKey: "MeetingLink")
+        
+        if (savedMeetingLink != nil)
+        {
+            meetingLink = savedMeetingLink as! String
+        }
+    }
+    
+    private func GetPermissions() async -> Void
+    {
+        if (token.isEmpty || token == "ACS token")
+        {
+            await ShowMessage(message: "Invalid token")
+            return
+        }
+        
+        let cameraAuthorizationStatus = AVCaptureDevice.authorizationStatus(for: .video)
+        let microphoneAuthorizationStatus = AVCaptureDevice.authorizationStatus(for: .audio)
+
+        if cameraAuthorizationStatus == .notDetermined
+        {
+            AVCaptureDevice.requestAccess(for: .video, completionHandler: { (granted: Bool) in
+                
+            })
+        }
+        
+        if microphoneAuthorizationStatus == .notDetermined
+        {
+            AVCaptureDevice.requestAccess(for: .audio, completionHandler: { (granted: Bool) in
+                
+            })
+        }
+        
+        if (cameraAuthorizationStatus != .authorized || microphoneAuthorizationStatus != .authorized)
+        {
+            await ShowMessage(message: "you need to provide the camera and microphone permissions")
+            return
+        }
+        
+        await InitResources()
+    }
+
+    private func CreateAgent() async -> Void
+    {
+        do
+        {
+            let credential = try CommunicationTokenCredential(token: token)
+            callClient = CallClient()
+            let callAgentOptions = CallAgentOptions()
+            callAgentOptions.displayName = "iOS Quickstart User"
+            
+            callAgent = try await callClient!.createCallAgent(userCredential: credential, options: callAgentOptions)
+
+            deviceManager = try await callClient!.getDeviceManager()
+            
+            let defaults = UserDefaults.standard
+            defaults.set(token, forKey: "Token")
+        }
+        catch let ex
+        {
+            await ShowMessage(message: "Failed to create call agent")
+            print(ex.localizedDescription)
+        }
```

Comment:
Nit: swift creates an implicit `error` variable 
```suggestion
        catch {
            await ShowMessage(message: "Failed to create call agent")
            print(error.localizedDescription)
        }
```

Reply:
done